plugins/cursecheck.cpp

```diff
@@ -153,65 +138,36 @@ command_result cursecheck (color_ostream &out, vector <string> & parameters)
     bool verbose = false;
     size_t cursecount = 0;
 
-    for(size_t i = 0; i < parameters.size();i++)
+    for(auto parameter : parameters)
     {
-        if(parameters[i] == "help" || parameters[i] == "?")
-        {
-            out.print(  "  Search for cursed creatures (ghosts, vampires, necromancers, zombies, werebeasts).\n"
-                        "  With map cursor active only the current tile will be checked.\n"
-                        "  Without an in-game cursor the whole map/world will be scanned.\n"
-                        "  By default cursed creatures are only counted to make it more interesting.\n"
-                        "  By default dead and passive creatures (aka really dead) are ignored.\n"
-                        "Options:\n"
-                        "  detail  - show details (name and age shown ingame might differ)\n"
-                        "  ids     - add creature and race IDs to be show on the  output\n"
-                        "  nick    - try to set cursetype as nickname (does not always work)\n"
-                        "  all     - include dead and passive creatures\n"
-                        "  verbose - show all curse tags (if you really want to know it all)\n"
-                        );
-            return CR_OK;
-        }
-        if(parameters[i] == "detail")
+        if(parameter == "help" || parameter == "?")
+            return CR_WRONG_USAGE;
+        if(parameter == "detail")
             giveDetails = true;
-        if(parameters[i] == "ids")
+        if(parameter == "ids")
             giveUnitID = true;
-        if(parameters[i] == "nick")
+        if(parameter == "nick")
             giveNick = true;
-        if(parameters[i] == "all")
+        if(parameter == "all")
             ignoreDead = false;
-        if(parameters[i] == "verbose")
+        if(parameter == "verbose")
         {
             // verbose makes no sense without enabling details
             giveDetails = true;
             verbose = true;
         }
     }
 
-    // check whole map if no cursor is active
-    bool checkWholeMap = false;
-    if(cursorX == -30000)
+    // check whole map if no unit is selected
+    for(df::unit *unit : selected_unit ? vector{ selected_unit } : world->units.all)
```

Comment:
```suggestion
    for(df::unit *unit : selected_unit ? vector{ selected_unit } : world->units.active)
```

Reply:
I don't think there's any benefit to scanning non-active units

Reply:
I also don't see the point of doing so. However, it is part of the explicitly advertised functionality. Are you suggesting to remove the `all` option and everything that goes along with it?

Reply:
you're right -- I didn't see the phrasing "Include dead and passive cursed creatures". that does indeed mean `units.all`